src/utils.js

```diff
@@ -310,4 +330,56 @@ export const convertSubfolderArray = (folderOrderArray, rawFolderContents, subfo
 export const generateImageorFilePath = (customPath, fileName) => {
   if (customPath) return encodeURIComponent(`${customPath}/${fileName}`)
   return fileName
+}
+
+const b64toBlob = (b64Data, contentType='', sliceSize=512) => {
+  const byteCharacters = atob(b64Data);
+  const byteArrays = [];
+
+  for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
+    const slice = byteCharacters.slice(offset, offset + sliceSize);
+
+    const byteNumbers = new Array(slice.length);
+    for (let i = 0; i < slice.length; i++) {
+      byteNumbers[i] = slice.charCodeAt(i);
+    }
+
+    const byteArray = new Uint8Array(byteNumbers);
+    byteArrays.push(byteArray);
+  }
+
+  const blob = new Blob(byteArrays, {type: contentType});
+  return blob;
+}
+
+/**
+ * Checks if the current repo with siteName is private
+ * If repo is public, returns the raw GitHub image URL
+ * If repo is private, calls the backend image API endpoint to retrieve the b64 encoded image blob text and returns the blob URL
+ *
+ * @param {string} siteName - Name of Isomer page repo
+ * @param {string} filePath - File path of image in repo. Should be of the format '/images/folder/subfolder1/subfolder2.../imagename.ext'.
+ *    The leading '/' is optional. The filePath parameter should be URI decoded. Examples:
+ *    images/test-folder/image sample.png
+ *    /images/test.svg
+ *    /images/folder 1/folder2/folder3/names.jpg
+ * @returns {Promise<string>}
+ */
+export async function fetchImageURL(siteName, filePath) {
+  const cleanPath = filePath.replace(/^\//, '') //Remove leading / if it exists e.g. /images/example.png -> images/example.png
+  //If the image is public, return the link to the raw file, otherwise make a call to the backend API to retrieve the image blob
+  if (!window.sitesIsPrivate[siteName]) {
```

Comment:
Should use `localStorage.getItem()` to retrieve the map. 

Also let's remove this storage item when the user logs out in `LoginContext.jsx` (`logout` function)